pkg/prom/instance/instance.go

```diff
@@ -435,7 +513,7 @@ func (i *Instance) newDiscoveryManager(ctx context.Context) (*discoveryService,
 	// TODO(rfratto): refactor this to a function?
 	// TODO(rfratto): ensure job name name is unique
 	c := map[string]sd_config.ServiceDiscoveryConfig{}
-	for _, v := range i.cfg.ScrapeConfigs {
+	for _, v := range i.Config().ScrapeConfigs {
```

Comment:
Just for my understanding, we would only take the lock once - right?

Reply:
Yep, we'd take the lock to get the value of ScrapeConfigs prior to the loop, and then release it before iteration of the loop starts.